### PR TITLE
fix(lambda): validate logLevel with logFormat for advanced logging

### PR DESCRIPTION
**Pull request by go-to-k**
This PR adds to validate whether `applicationLogLevel` and `systemLogLevel` have a JSON `logFormat` for advanced logging.

The doc states the following:

> To use log-level filtering, your function must be configured to use the JSON log format. The default log format for all Lambda managed runtimes is currently plain text. To learn how to configure your function's log format to JSON, see [Setting your function's log format](https://docs.aws.amazon.com/lambda/latest/dg/monitoring-cloudwatchlogs.html#monitoring-cloudwatchlogs-set-format).

https://docs.aws.amazon.com/lambda/latest/dg/monitoring-cloudwatchlogs.html#monitoring-cloudwatchlogs-advanced

If you specify `systemLogLevel` or `applicationLogLevel` without `LogFormat.JSON` like the following code, an error occurred in a CFn event: 
```
Resource handler returned message: "LogLevel is not supported when LogFormat is set to "Text". Remove LogLevel from your request or change the LogFormat to "Json" and try again.
```

```ts
new Function(stack, 'LambdaWithLogLevel', {
  code: new InlineCode('foo'),
  handler: 'index.handler',
  runtime: Runtime.NODEJS_18_X,
  logFormat: LogFormat.TEXT,
  systemLogLevel: SystemLogLevel.INFO,
  applicationLogLevel: ApplicationLogLevel.INFO,
});
```

This validation allows the error to be caught at the synth phase.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-lambda/lib/function.ts [hunk 1]
```diff
@@ -1077,6 +1077,10 @@ export class Function extends FunctionBase {
    * function and undefined if not.
    */
   private getLoggingConfig(props: FunctionProps): CfnFunction.LoggingConfigProperty | undefined {
+    if ((props.applicationLogLevel || props.systemLogLevel) && props.logFormat !== LogFormat.JSON) {
+      throw new Error('ApplicationLogLevel and SystemLogLevel cannot be specified without LogFormat being set to JSON.');
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-lambda/lib/function.ts:1081]:
```suggestion
      throw new Error(`To use ApplicationLogLevel and/or SystemLogLevel you must set LogFormat to '${LogFormat.JSON}', got '${props.logFormat}'.`);
```
Message formatting.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍
Just a small adjustment on the message in my opinion.
Also, can you please add a note [here](https://github.com/aws/aws-cdk/blob/47135471abd0ea7472ff160676310e0f7eab8d89/packages/aws-cdk-lib/aws-lambda/README.md?plain=1#L175) on this requirement?

**Comment by go-to-k** [comment]
@lpizzinidev 

Thanks! I changed it.

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Review by sumupitchayan — DISMISSED** [review]
Thanks for your contribution!

**Comment by vinayak-kukreja** [comment]
@Mergifyio update

**Review by sumupitchayan — APPROVED** [review]

**Pull request merged**